### PR TITLE
support npm dependency aliasing

### PR DESCRIPTION
**Pull request by alexlamsl**

### CODE REVIEW

File: src/install/install.zig [hunk 1]
```diff
@@ -1458,6 +1375,7 @@ pub const PackageManager = struct {
     resolve_tasks: TaskChannel,
     timestamp_for_manifest_cache_control: u32 = 0,
     extracted_count: u32 = 0,
+    aliasMap: std.ArrayHashMapUnmanaged(PackageID, String, ArrayIdentityContext, false) = .{},
```
Comment (Jarred-Sumner) [inline src/install/install.zig]: this should technically be `alias_map` to fit the zig convention of variable names being snake_case though we don't follow that consistently (particularly with globalObject or globalThis where we want to sort of match JavaScript convention)

File: src/install/install.zig [hunk 2]
```diff
@@ -2234,64 +2155,6 @@ pub const PackageManager = struct {
         return network_task;
     }
 
-    pub fn fetchVersionsForPackageName(
```
Comment (Jarred-Sumner) [inline src/install/install.zig:2237]: was this dead code?

File: src/install/install.zig (hunk 1 above)
Comment (alexlamsl) [reply src/install/install.zig]: Ah oops - will fix now.

File: src/install/install.zig (hunk 2 above)
Comment (alexlamsl) [reply src/install/install.zig:2237]: Yes.

File: src/install/install.zig (hunk 1 above)
Comment (alexlamsl) [reply src/install/install.zig]: Done in https://github.com/oven-sh/bun/pull/1837/commits/1acbdcb40502a341c47a80f9ccfcb3d97c42e7d2

**Review by Jarred-Sumner — APPROVED** [review]
When we install the aliased version, it installs into the aliased folder inside node_modules? 

When saving it into the lockfile, there are no changes because we check the npm: prefix and from that parse out the alias or non-alias?

**Comment by alexlamsl** [comment]
> When we install the aliased version, it installs into the aliased folder inside node_modules?

Aliased folder name inside `./node_modules`, original package name inside `.bun/install/cache`

> When saving it into the lockfile, there are no changes because we check the npm: prefix and from that parse out the alias or non-alias?

The `npm:` prefix is stored within `Dependency.Version.literal`, which should get (de)serialised and re-parsed by `Dependency.parseWithTag()` accordingly.

**Pull request merged**